app/background-process/api-manifests/browser.js

```diff
@@ -14,5 +14,9 @@ export default {
   uninstallPlugin: 'promise',
 
   getHomePages: 'promise',
-  getProtocolDescription: 'sync'
-}
\ No newline at end of file
+  getProtocolDescription: 'sync',
+
+  getDefaultProtocolSettings: 'sync',
+  setAsDefaultProtocolClient: 'sync',
+  removeAsDefaultProtocolClient: 'sync'
```

Comment:
Lets make these RPC calls promises instead of sync. Even sync calls freeze the thread briefly, and, if they fail, can freeze up the program -- so I try to avoid them.